src/msg/async/frames_v2.h

```diff
@@ -91,8 +104,30 @@ struct preamble_block_t {
 static_assert(sizeof(preamble_block_t) % CRYPTO_BLOCK_SIZE == 0);
 static_assert(std::is_standard_layout<preamble_block_t>::value);
 
+// Each Frame has an epilogue for integrity or authenticity validation.
+// For plain mode it's quite straightforward - the structure stores up
+// to MAX_NUM_SEGMENTS crc32 checksums, one per each segment.
+// For secure mode things become very different. The fundamental thing
```

Comment:
Basically plain mode can be seen as special case of secure mode: encryptor doing `cipertext := plaintext` + very weak message authentication code (but with some extra integrity facilities). The sole difference I see at the moment is absence of the `CRYPTO_BLOCK_SIZE` rounding. This has potential for unification (rather post-nautilus).